src/Core/Systems/ResourceSystem.hh

```diff
@@ -0,0 +1,76 @@
+#ifndef ESPERT_CORE_SYSTEMS_RESOURCESYSTEM_HH
+#define ESPERT_CORE_SYSTEMS_RESOURCESYSTEM_HH
+
+#include "Core/Resources/ResourceTypes.hh"
+#include "esppch.hh"
+
+namespace esp
+{
+  class Loader
+  {
+   public:
+    Loader()  = default;
+    ~Loader() = default;
+
+    virtual std::unique_ptr<Resource> load(const fs::path& path, const ResourceParams& params) = 0;
+    virtual void unload(std::unique_ptr<Resource> resource)                                    = 0;
+  };
+
+  // type map implementation from https://en.cppreference.com/w/cpp/types/type_info/hash_code
+  using TypeInfoRef = std::reference_wrapper<const std::type_info>;
+
+  struct Hasher
+  {
+    std::size_t operator()(TypeInfoRef code) const { return code.get().hash_code(); }
+  };
+
+  struct EqualTo
+  {
+    bool operator()(TypeInfoRef lhs, TypeInfoRef rhs) const { return lhs.get() == rhs.get(); }
+  };
+
+  using LoaderMap = std::unordered_map<TypeInfoRef, std::unique_ptr<Loader>, Hasher, EqualTo>;
+
+  class ResourceSystem
+  {
+   public:
+    PREVENT_COPY(ResourceSystem);
+
+    static std::unique_ptr<ResourceSystem> init(const fs::path& asset_base_path);
```

Comment:
Jako że wszędzie w render api używamy nazw "create" i "terminate" to może i tutaj by lepiej takich użyć dla spójności? Nie upieram się to tylko sugestia

Reply:
Dobry pomysł, poprawiłem.
Co do połączenia tego resource systemu z resource managerem w render api to aktualnie nad tym pracuję przy okazji texture systemu i efekty powinny pojawić się niedługo ;)